features/contain-style.yml

```
@@ -0,0 +1,6 @@
+name: Style containment
+description: "The `contain: style` CSS declaration TODO."
```

```suggestion
description: "The `contain: style` CSS declaration prevents counter (`counter-increment` and `counter-set`) and quotation styles (`content` property quote values) from applying beyond the element's descendants. This permits the browser to avoid slower rendering calculations.
```

---

@danielsakhapov I think you know about this, can you say if the description of this feature is accurate?

---

I'd say it prevents modifying counters and quotes values outside the element's descendants, but doesn't prevent accessing the value

---

This is a little bit hard to understand. Do web developers use style containment because they care about counters and quotes, or do they really use it because it makes stuff faster without them knowing why? If so, maybe we should say that it makes stuff faster, and that the side effect is on counters and quotes.

---

How about this, to shift the emphasis to something like "go faster, at a specific expense"?

```suggestion
description: "The `contain: style` CSS declaration permits the browser to avoid slower layout calculations by preventing modification to counter (`counter-increment` and `counter-set`) and quotation styles (`content` property quote values) beyond the element's descendants.
```